graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,205 @@
+
+    R: int
+        Number of restarts for soft seeded graph matching algorithm
+
+    a_inds: 1d-array
+        Array of indices in the induced subgraph of A that will be matched to `b_inds`
+
+    b_inds: 1d-array
+        Array of indices in the induced subgraph of B that will be matched to `a_inds`
+
+    n_seeds_used: int
+        Number of seeds passed in `seedsA` that occured in the induced subgraph about `voi`
+
+    nomination_list: list
+        List of 2 tuples of the format nomination_list = [(j, p_val)] where p_val is the
+        probability that the voi matches to node j in graph B (sorted by descending
+        probability)
+
+    Attributes
+    ----------
+    P : 2d-array, square
+        The probability array that describes the probability `P[i, j]` that row `i` in `a_inds`
+        matches `j` in `b_inds`
+
+    corr : 1d-array
+        The matrix that maps values from `a_inds` to `b_inds`
+
+
+    References
+    ----------
+    .. [1] Patsolic, HG, Park, Y, Lyzinski, V, Priebe, CE. Vertex nomination via seeded graph matching. Stat Anal Data
+    Min: The ASA Data Sci Journal. 2020; 13: 229– 244. https://doi.org/10.1002/sam.11454
+    """
+
+    def __init__(
+        self,
+        h=1,
+        ell=1,
+        R=100
+    ):
+        self.h = h
+        self.ell = ell
+        self.R = R
+
+    def fit(self, voi, A, B, seedsA, seedsB):
+        """
+        Parameters
+        ----------
+
+        voi: int
+            Vertices of interest (voi)
+
+        A: 2d-array, square
+            Adjacency matrix of `G_1`, the graph where voi is known
+
+        B: 2d-array, square
+            Adjacency matrix of `G_2`, the graph where voi is not known
+
+        seedsA: list
+            Seeds associated to adjacency matrix A
+
+        seedsB: list
+            Seeds associated to adjacency matrix B `len(seedsA)==len(seedsB)`
+        """
+        assert len(seedsA) == len(seedsB)
+
+        voi = np.reshape(np.array(voi), (1,))
+
+        # get reordering for A
+        nsx1 = np.setdiff1d(list(range(A.shape[0])), np.concatenate((seedsA, voi)))
+        a_reord = np.concatenate((seedsA, voi, nsx1))
+
+        # get reordering for B
+        nsx2 = np.setdiff1d(list(range(B.shape[0])), seedsB)
+        b_reord = np.concatenate((seedsB, nsx2))
+
+        AA = A[np.ix_(a_reord, a_reord)]
+        BB = B[np.ix_(b_reord, b_reord)]
+
+        seeds_reord = np.arange(len(seedsA))
+        voi_reord = len(seedsA)
+        #     print("seeds reord = ", seeds_reord, ", voi_reord = ", voi_reord)
+
+        subgraph_AA = np.array(ego_list(AA, self.h, voi_reord, mindist=1))
+
+        voi_reord = np.reshape(np.array(voi_reord), (1,))
+
+        Sx1 = np.intersect1d(a_reord[subgraph_AA], seeds_reord)
+        Sx2 = np.intersect1d(np.arange(BB.shape[0]), Sx1)
+        #     print("Sx1 = ", Sx1, " Sx2 = ", Sx2)
+
+        if len(Sx2) <= 0:
+            print("Impossible")
```

Comment:
raise a TypeException here.

Reply:
I cant raise an exception here because the algorithm didn't fail. This is a condition where the algorithm cannot work, I think none is the best thing to return.